In [6]:
import pandas as pd
import numpy as np

from sqlalchemy import create_engine

import plotly.express as px
import cufflinks as cf

cf.go_offline()

In [7]:
from sqlalchemy import create_engine
from sqlalchemy.engine import URL

connection_url = URL.create(
    "mysql+pymysql",
    username="root",
    password="Jay@1234",
    host="localhost",
    port=3306,
    database="ecommerce_project"
)

engine = create_engine(connection_url)

In [8]:
import pandas as pd

users = pd.read_sql("select * from users limit 5;", engine)
users

,user_id,name,email,gender,city,signup_date
0,U000001,Angel Hill,donaldgarcia@example.net,Other,New Roberttown,2025-03-13
1,U000002,Jesse Guzman,jennifermiles@example.com,Male,South Bridget,2024-03-05
2,U000003,Adam Shaffer,jpeterson@example.org,Male,Curtisfurt,2025-07-07
3,U000004,Melanie Munoz,blairamanda@example.com,Other,New Kellystad,2024-03-07
4,U000005,Janet Williams,kendragalloway@example.org,Female,South Joshuastad,2025-01-29


In [9]:
users = pd.read_sql("select * from users;", engine)

products = pd.read_sql("select * from products;", engine)

orders = pd.read_sql("select * from orders;", engine)

order_items = pd.read_sql("select * from order_items;", engine)

reviews = pd.read_sql("select * from reviews;", engine)

events = pd.read_sql("select * from events;", engine)

In [10]:
print(users.shape)
print(products.shape)
print(orders.shape)
print(order_items.shape)
print(reviews.shape)
print(events.shape)

(10000, 6)
(2000, 6)
(20000, 5)
(43525, 7)
(15000, 7)
(80000, 5)


In [11]:
users.head()

,user_id,name,email,gender,city,signup_date
0,U000001,Angel Hill,donaldgarcia@example.net,Other,New Roberttown,2025-03-13
1,U000002,Jesse Guzman,jennifermiles@example.com,Male,South Bridget,2024-03-05
2,U000003,Adam Shaffer,jpeterson@example.org,Male,Curtisfurt,2025-07-07
3,U000004,Melanie Munoz,blairamanda@example.com,Other,New Kellystad,2024-03-07
4,U000005,Janet Williams,kendragalloway@example.org,Female,South Joshuastad,2025-01-29


In [17]:
fashion_sales = pd.read_csv("fashion_sales.csv")

In [20]:
fashion_sales.shape

(4605, 25)

# Fashion Sales Dataset – Feature Preparation

The `fashion_sales` dataset was extracted from the original e-commerce dataset and contains sales information related to clothing products.

The original fashion dataset contained **4,605 rows and 22 columns**. During the data preparation and analysis stages, additional features were created to support the business analysis and machine learning recommendation system.

## Additional Features Created

### 1. Order Month

The `order_month` column was created from the `order_date` column.

This feature represents the month in which an order was placed and is useful for analyzing sales and revenue trends over time.

For example:

- `2025-01` → January 2025
- `2025-02` → February 2025
- `2025-03` → March 2025

The `order_month` feature was used for **Monthly Revenue Analysis**.



### 2. Price Range

The `price_range` column was created from the product price to group products into different price categories.

The price ranges help analyze how product prices affect sales and customer purchasing behavior.

The categories include:

- Under $50
- $50-$100
- $100-$200
- $200-$500
- Above $500

The `price_range` feature was used for **Product Sales by Price Range** analysis.



### 3. Subcategory

The original dataset contains only one main category: **Clothing**.

However, the dataset does not contain specific clothing types such as T-Shirts, Jeans, Jackets, or Hoodies. Therefore, a new `subcategory` feature is introduced as an engineered feature to support the personalized fashion recommendation system.

The subcategories include:

- T-Shirt
- Shirt
- Jeans
- Jacket
- Hoodie
- Dress
- Sweater
- Shorts
- Skirt
- Tracksuit

This feature allows the recommendation system to provide more specific clothing recommendations.

## Dataset Evolution

The dataset changed as new features were engineered:

**Original Dataset:**  
4,605 rows × 22 columns

**After adding `order_month` and `price_range`:**  
4,605 rows × 24 columns

**After adding `subcategory`:**  
4,605 rows × 25 columns

The number of rows remains unchanged because these steps only add new features to existing records.

## Purpose of Feature Engineering

Feature engineering helps transform existing data into useful features for analysis and machine learning.

The additional features allow the project to analyze:

- Monthly sales and revenue trends
- Product sales across different price ranges
- Clothing subcategory performance
- Customer preferences
- More personalized product recommendations

These engineered features will later be used in the **AI Personalized Fashion Shopping Assistant**.

In [21]:
fashion_sales.columns.tolist()

['order_id',
 'user_id',
 'order_date',
 'order_status',
 'total_amount',
 'name',
 'email',
 'gender',
 'city',
 'signup_date',
 'order_item_id',
 'product_id',
 'quantity',
 'item_price',
 'item_total',
 'product_name',
 'category',
 'brand',
 'price',
 'product_rating',
 'review_rating',
 'review_text',
 'order_month',
 'price_range',
 'subcategory']

In [22]:
fashion_sales[[
    "order_month",
    "price_range",
    "subcategory"
]].head()

,order_month,price_range,subcategory
0,2025-10,Mid Range ($50-$200),Jeans
1,2024-10,Premium ($200-$500),Skirt
2,2025-10,Mid Range ($50-$200),Dress
3,2025-08,Mid Range ($50-$200),Jacket
4,2025-08,Budget ($0-$50),T-Shirt


In [31]:
fashion_sales.rename(
    columns={"subcategory": "product_type"},
    inplace=True
)

In [33]:
fashion_sales.columns.tolist()

['order_id',
 'user_id',
 'order_date',
 'order_status',
 'total_amount',
 'name',
 'email',
 'gender',
 'city',
 'signup_date',
 'order_item_id',
 'product_id',
 'quantity',
 'item_price',
 'item_total',
 'product_name',
 'category',
 'brand',
 'price',
 'product_rating',
 'review_rating',
 'review_text',
 'order_month',
 'price_range',
 'product_type']

In [34]:
fashion_sales[["product_name", "category", "product_type"]].head(10)

,product_name,category,product_type
0,Nimbus When,Clothing,Jeans
1,Harbor Night,Clothing,Skirt
2,Astra Without,Clothing,Dress
3,Orion Most,Clothing,Jacket
4,Nimbus Sit,Clothing,T-Shirt
5,Harbor Beat,Clothing,T-Shirt
6,Solace Car,Clothing,T-Shirt
7,Astra Test,Clothing,Sweater
8,Harbor Night,Clothing,Hoodie
9,Acme Wrong,Clothing,Dress


In [38]:
fashion_sales.to_csv("fashion_sales.csv", index=False)

In [39]:
fashion_sales["price_range"].value_counts()

price_range
Mid Range ($50-$200)    3078
Budget ($0-$50)         1004
Premium ($200-$500)      523
Name: count, dtype: int64

In [40]:
fashion_sales["order_month"].value_counts().sort_index()

order_month
2024-01    211
2024-02    186
2024-03    216
2024-04    209
2024-05    195
2024-06    210
2024-07    206
2024-08    207
2024-09    197
2024-10    205
2024-11    198
2024-12    199
2025-01    223
2025-02    181
2025-03    209
2025-04    196
2025-05    225
2025-06    185
2025-07    200
2025-08    203
2025-09    219
2025-10    230
2025-11     95
Name: count, dtype: int64

### Visualize subcategory distribution

In [42]:
product_type_count = (
    fashion_sales["product_type"]
    .value_counts()
    .reset_index()
)

product_type_count.columns = ["product_type", "count"]

fig = px.bar(
    product_type_count,
    x="product_type",
    y="count",
    color="count",
    title="Distribution of Clothing Product Types",
    labels={
        "product_type": "Product Type",
        "count": "Number of Records"
    }
)

fig.show()

# Business Analysis for Subcategory

### Revenue by product_type

In [43]:
product_type_revenue = (
    fashion_sales
    .groupby("product_type")["item_total"]
    .sum()
    .sort_values(ascending=False)
    .reset_index()
)

fig = px.bar(
    product_type_revenue,
    x="product_type",
    y="item_total",
    color="item_total",
    title="Revenue by Product Type",
    labels={
        "product_type": "Product Type",
        "item_total": "Revenue ($)"
    }
)

fig.show()

### Best-Selling Product Types

In [44]:
product_type_sales = (
    fashion_sales
    .groupby("product_type")["quantity"]
    .sum()
    .sort_values(ascending=False)
    .reset_index()
)

fig = px.bar(
    product_type_sales,
    x="product_type",
    y="quantity",
    color="quantity",
    title="Best-Selling Product Types",
    labels={
        "product_type": "Product Type",
        "quantity": "Units Sold"
    }
)

fig.show()

### Average Rating by Product Type

In [45]:
product_type_rating = (
    fashion_sales
    .groupby("product_type")["product_rating"]
    .mean()
    .sort_values(ascending=False)
    .reset_index()
)

fig = px.bar(
    product_type_rating,
    x="product_type",
    y="product_rating",
    color="product_rating",
    title="Average Product Rating by Product Type",
    labels={
        "product_type": "Product Type",
        "product_rating": "Average Rating"
    }
)

fig.show()

In [46]:
fashion_sales.to_csv("fashion_sales.csv", index=False)

In [47]:
fashion_sales.to_csv(
    "/Users/jayanthkumar/Desktop/Ecommerce_project/jupyter_nb/Ecommerce_Dashboard/fashion_sales.csv",
    index=False
)

In [48]:
fashion_sales.shape

(4605, 25)

In [49]:
fashion_sales.columns.tolist()

['order_id',
 'user_id',
 'order_date',
 'order_status',
 'total_amount',
 'name',
 'email',
 'gender',
 'city',
 'signup_date',
 'order_item_id',
 'product_id',
 'quantity',
 'item_price',
 'item_total',
 'product_name',
 'category',
 'brand',
 'price',
 'product_rating',
 'review_rating',
 'review_text',
 'order_month',
 'price_range',
 'product_type']